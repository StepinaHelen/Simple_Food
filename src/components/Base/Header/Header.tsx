import React, { useContext } from "react";
import Icons from "components/SvgComponent/SvgComponent";
import CommonContainer from "components/Base/Containers/CommonContainer";
import { Link } from "react-router-dom";
import CartContext from "store/cart-context";
import { ROUTES } from "common/constants";
import {
  HeaderContainer,
  HeaderWrapper,
  Title,
  IconsContainer,
  Amount,
  IconContainer,
  ThemeBtn,
} from "./HeaderStyles";

const Header = (props: any) => {
  const cartContext = useContext(CartContext);
  const cartItemsAmount = cartContext.items.reduce((currentValue, item) => {
    return currentValue + item.amount;
  }, 0);
  return (
    <React.Fragment>
      {props.mode === "dark" ? (
        <ThemeBtn onClick={props.changeTheme}>
          <Icons name="moon" classes={"theme-moon"} />
        </ThemeBtn>
      ) : (
        <ThemeBtn onClick={props.changeTheme}>
          <Icons name="sun" classes={"theme"} />
        </ThemeBtn>
      )}
      <HeaderContainer>
        <CommonContainer withMargin={false}>
          <HeaderWrapper>
            <Link to={"/"}>
              <Icons name="logo" classes={"icon"} />
            </Link>
            <Title>Simple food</Title>
            <IconsContainer>
              <Link to={ROUTES.cartPage}>
                <IconContainer>
                  <Icons name="cart" classes={"icon"} />
                  <Amount>
                    <span>{cartItemsAmount}</span>
                  </Amount>
                </IconContainer>
              </Link>
              <hr />
              <Link to={ROUTES.orderHistoryPage}>
                <Icons name="note" classes={"icon"} />
              </Link>
            </IconsContainer>
          </HeaderWrapper>
        </CommonContainer>
      </HeaderContainer>
    </React.Fragment>
  );
};

export default Header;
