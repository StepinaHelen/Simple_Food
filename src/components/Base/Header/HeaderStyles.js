import styled from "styled-components";

export const HeaderContainer = styled.header`
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  background-color: #6b068a;
  color: white;
  box-shadow: 0 2px 10px rgba(0, 0, 0, 0.25);
  z-index: 10;
`;

export const HeaderWrapper = styled.div`
  margin: 0 auto;
  display: flex;
  padding: 12px 0;
  justify-content: space-between;
  align-items: center;
  color: white;

  .icon {
    height: 62px;
    width: 48px;
  }
`;

export const Title = styled.div`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  font-family: "Mansalva", cursive;
  font-weight: 400;
  font-size: 52px;
  line-height: 57px;

  @media screen and (max-width: 768px) {
    font-weight: 400;
    font-size: 32px;
    line-height: 35px;
  }
`;

export const IconsContainer = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  hr {
    transform: rotate(180deg);
    height: 56px;
    margin-right: 8px;
    margin-left: 8px;
    border: 2px solid #ffffff;
    background-color: white;
    border-radius: 3px;
    @media screen and (max-width: 768px) {
      margin-right: 2px;
      margin-left: 2px;
      height: 40px;
    }
  }
`;

export const Amount = styled.div`
  position: absolute;
  bottom: 10px;
  right: 0px;
  width: 25px;
  height: 25px;
  border-radius: 50%;
  background-color: #350e42;

  span {
    display: block;
    color: white;
    font-weight: 600;
    margin-left: 8px;
    font-size: 18px;
  }
`;

export const IconContainer = styled.div`
  position: relative;
`;
