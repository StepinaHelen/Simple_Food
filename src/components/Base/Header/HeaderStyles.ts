import styled from "styled-components";

export const HeaderContainer = styled.header`
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  background-color: ${({ theme }) => theme.colors.primary};
  color: white;
  box-shadow: 0 2px 10px rgba(0, 0, 0, 0.25);
  z-index: 10;
`;

export const HeaderWrapper = styled.div`
  ${({ theme }) =>
    theme.flex({
      alignItems: "center",
      justifyContent: "space-between",
    })}
  margin: 0 auto;
  padding: 12px 0;
  color: white;
  .icon {
    height: 62px;
    width: 48px;
  }
`;

export const Title = styled.div`
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  font-family: "Mansalva", cursive;
  ${({ theme }) => theme.fontText({ fontSize: 52, lineHeight: 57 })}

  @media screen and (max-width: 768px) {
    ${({ theme }) => theme.fontText({ fontSize: 32, lineHeight: 35 })}
  }
`;

export const IconsContainer = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  hr {
    transform: rotate(180deg);
    height: 56px;
    margin-right: 8px;
    margin-left: 8px;
    border: 2px solid ${({ theme }) => theme.colors.secondaryFont};
    background-color: white;
    border-radius: 3px;
    @media screen and (max-width: 768px) {
      margin-right: 2px;
      margin-left: 2px;
      height: 40px;
    }
  }
`;

export const Amount = styled.div`
  position: absolute;
  bottom: 10px;
  right: 0px;
  width: 25px;
  height: 25px;
  border-radius: 50%;
  background-color: #350e42;

  span {
    display: block;
    color: white;
    font-weight: 600;
    margin-left: 8px;
    font-size: 18px;
    margin-top: 3px;
  }
`;

export const IconContainer = styled.div`
  position: relative;
`;

export const ThemeBtn = styled.button`
  position: fixed;
  z-index: 1000;
  top: 25px;
  right: 25px;
  background-color: transparent;
  border: none;
  .theme {
    height: 48px;
    width: 48px;
    &-moon {
      height: 40px;
      width: 40px;
    }
  }
`;
