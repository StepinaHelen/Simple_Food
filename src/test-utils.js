import { render } from "@testing-library/react";
import { QueryClientProvider, QueryClient } from "react-query";
import { BrowserRouter } from "react-router-dom";
import { ThemeProvider } from "styled-components";
import { baseTheme } from "./styles/theme";
const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
    },
  },
});
const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={baseTheme}>
      <QueryClientProvider contextSharing={true} client={queryClient}>
        <BrowserRouter>{children}</BrowserRouter>
      </QueryClientProvider>
    </ThemeProvider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";

export { customRender as render };
