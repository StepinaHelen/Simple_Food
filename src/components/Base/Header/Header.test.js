import { render, screen } from "test-utils";
import Header from "components/Base/Header/Header";
describe("test Footer", () => {
  it("Header renders", () => {
    render(<Header />);
  });
  it("Header snapshot", () => {
    const view = render(<Header />);
    expect(view).toMatchSnapshot();
  });

  it("check Header text", () => {
    render(<Header />);
    expect(screen.getByText("Simple food")).toBeInTheDocument();
  });
});
