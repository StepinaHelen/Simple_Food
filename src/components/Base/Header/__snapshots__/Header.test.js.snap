// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Footer Header snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c1 {
  padding-left: 15px;
  padding-right: 15px;
  margin-left: auto;
  margin-right: auto;
}

.c0 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  background-color: #6b068a;
  color: white;
  box-shadow: 0 2px 10px rgba(0,0,0,0.25);
  z-index: 10;
}

.c2 {
  margin: 0 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 12px 0;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: white;
}

.c2 .icon {
  height: 62px;
  width: 48px;
}

.c3 {
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  font-family: "Mansalva",cursive;
  font-weight: 400;
  font-size: 52px;
  line-height: 57px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 hr {
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
  height: 56px;
  margin-right: 8px;
  margin-left: 8px;
  border: 2px solid #ffffff;
  background-color: white;
  border-radius: 3px;
}

.c6 {
  position: absolute;
  bottom: 10px;
  right: 0px;
  width: 25px;
  height: 25px;
  border-radius: 50%;
  background-color: #350e42;
}

.c6 span {
  display: block;
  color: white;
  font-weight: 600;
  margin-left: 8px;
  font-size: 18px;
}

.c5 {
  position: relative;
}

@media screen and (min-width:480px) {
  .c1 {
    width: 480px;
  }
}

@media screen and (min-width:768px) {
  .c1 {
    width: 768px;
  }
}

@media screen and (min-width:1200px) {
  .c1 {
    width: 1200px;
  }
}

@media screen and (max-width:768px) {
  .c3 {
    font-weight: 400;
    font-size: 32px;
    line-height: 35px;
  }
}

@media screen and (max-width:768px) {
  .c4 hr {
    margin-right: 2px;
    margin-left: 2px;
    height: 40px;
  }
}

<body>
    <div>
      <header
        class="c0"
      >
        <div
          class="c1"
        >
          <div
            class="c2"
          >
            <a
              href="/"
            >
              <svg
                class="icon"
              >
                <use
                  xlink:href="sprite.svg#logo"
                />
              </svg>
            </a>
            <div
              class="c3"
            >
              Simple food
            </div>
            <div
              class="c4"
            >
              <a
                href="/cart"
              >
                <div
                  class="c5"
                >
                  <svg
                    class="icon"
                  >
                    <use
                      xlink:href="sprite.svg#cart"
                    />
                  </svg>
                  <div
                    class="c6"
                  >
                    <span>
                      0
                    </span>
                  </div>
                </div>
              </a>
              <hr />
              <a
                href="/order-history"
              >
                <svg
                  class="icon"
                >
                  <use
                    xlink:href="sprite.svg#note"
                  />
                </svg>
              </a>
            </div>
          </div>
        </div>
      </header>
    </div>
  </body>,
  "container": .c1 {
  padding-left: 15px;
  padding-right: 15px;
  margin-left: auto;
  margin-right: auto;
}

.c0 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  background-color: #6b068a;
  color: white;
  box-shadow: 0 2px 10px rgba(0,0,0,0.25);
  z-index: 10;
}

.c2 {
  margin: 0 auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 12px 0;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: white;
}

.c2 .icon {
  height: 62px;
  width: 48px;
}

.c3 {
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  font-family: "Mansalva",cursive;
  font-weight: 400;
  font-size: 52px;
  line-height: 57px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 hr {
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
  height: 56px;
  margin-right: 8px;
  margin-left: 8px;
  border: 2px solid #ffffff;
  background-color: white;
  border-radius: 3px;
}

.c6 {
  position: absolute;
  bottom: 10px;
  right: 0px;
  width: 25px;
  height: 25px;
  border-radius: 50%;
  background-color: #350e42;
}

.c6 span {
  display: block;
  color: white;
  font-weight: 600;
  margin-left: 8px;
  font-size: 18px;
}

.c5 {
  position: relative;
}

@media screen and (min-width:480px) {
  .c1 {
    width: 480px;
  }
}

@media screen and (min-width:768px) {
  .c1 {
    width: 768px;
  }
}

@media screen and (min-width:1200px) {
  .c1 {
    width: 1200px;
  }
}

@media screen and (max-width:768px) {
  .c3 {
    font-weight: 400;
    font-size: 32px;
    line-height: 35px;
  }
}

@media screen and (max-width:768px) {
  .c4 hr {
    margin-right: 2px;
    margin-left: 2px;
    height: 40px;
  }
}

<div>
    <header
      class="c0"
    >
      <div
        class="c1"
      >
        <div
          class="c2"
        >
          <a
            href="/"
          >
            <svg
              class="icon"
            >
              <use
                xlink:href="sprite.svg#logo"
              />
            </svg>
          </a>
          <div
            class="c3"
          >
            Simple food
          </div>
          <div
            class="c4"
          >
            <a
              href="/cart"
            >
              <div
                class="c5"
              >
                <svg
                  class="icon"
                >
                  <use
                    xlink:href="sprite.svg#cart"
                  />
                </svg>
                <div
                  class="c6"
                >
                  <span>
                    0
                  </span>
                </div>
              </div>
            </a>
            <hr />
            <a
              href="/order-history"
            >
              <svg
                class="icon"
              >
                <use
                  xlink:href="sprite.svg#note"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
